### PR TITLE
feat: grant ebpf k8s collector job watching permission.

### PR DESCRIPTION
**Pull request by hyfj44255**
# Description

[opentelemetry-network](https://github.com/open-telemetry/opentelemetry-network) added [k8s collector handle pods owned by a CronJob](https://github.com/open-telemetry/opentelemetry-network/commit/cf154bd5144263a3d3db7e3963adf474c0a76372) feature so we need to grant EBPF k8s collector job watching permission in k8s-collector-clusterrole.

Fixes # (issue)

https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1349

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Go to open-telemetry-opentelemetry-helm-chartsrepo root directory. 

```cd opentelemetry-helm-charts```

Create a kind cluster same as the one in repo's Github action 

```kind create cluster --name ebpf-cluster --config ./.github/kind-1.24.yaml```

Go to EBPF chart 

```cd charts/opentelemetry-ebpf```

Run helm install to install the EBPF chart 

```helm install ebpf-chart-test . -f ci/requests-limits-values.yaml -n ebpf-chart-test```

Run command to get ebpf  k8s collector container -  k8s-watcher's log, and search 'jobs.batch is forbidden' , if not able to search any error related to "not able to list job" means this PR's changing is working

```
kubectl logs `kubectl get pod -n ebpf-chart-test |grep ebpf-chart-test-opentelemetry-ebpf-k8s-collector | \
awk -F ' ' '{print $1}'` -n ebpf-chart-test -c k8s-watcher | \
grep 'jobs.batch is forbidden'
```

### CODE REVIEW

**Comment by hyfj44255** [comment]
Hi @TylerHelmuth could you take a look this PR when you get chance ,it's related to adjust the helm chart according to the change of [Opentelemetry eBPF](https://github.com/open-telemetry/opentelemetry-ebpf/releases). https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1349. BTW the I'm working on fixing the CI failure

**Comment by hyfj44255** [comment]
Hi @TylerHelmuth the previous lint -test issue was fixed would it be okay if you help trigger the lint-test again?

**Comment by hyfj44255** [comment]
Hi @nicolastakashi I was wonder if you could take a look this PR, sorry that it contains some other change from my other PR, I will fix it latter, but would it be okay if you take a look it in advance? thanks!

File: charts/opentelemetry-ebpf/ci/enabled-log-debug-values.yaml [hunk 1]
```diff
@@ -10,4 +10,4 @@ log:
 debug:
   enabled: true
   storeMinidump: true
-  sendUnplannedExitMetric: false
```
Comment (nicolastakashi) [inline charts/opentelemetry-ebpf/ci/enabled-log-debug-values.yaml:13]: I think we should use same idea I mentioned on #1347 
let's make it a different test file

**Comment by hyfj44255** [comment]
Hi @nicolastakashi I've got rid of the changes not related to the issue this PR is trying to fix , I was wondering if you could take a look again, thanks!

**Review by nicolastakashi — APPROVED** [review]

**Comment by nicolastakashi** [comment]
LGTM 🥳 
Although I don't have merge permissions.
@tylerbenson can you help merging it, if it's looking good for you?

**Review by TylerHelmuth — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
@hyfj44255 I merged the other PR, can you check if this needs a rebase? I dont have permission to update your branch with the latest from main

**Comment by hyfj44255** [comment]
Hi @TylerHelmuth I've updated this branch with the latest from main, would it be okay if you take a look again?

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**